Tsats util import

FlyingMonkey4 opened the pull request:
Added TSAT Utilities submodules to the dev branch and confirmed that it compiles correctly, but haven't tested the code.

FlyingMonkey4 committed:
Added Submodules for TSAT Utilities

Added the submodules for both CDH projects and activated TIM16 for the CAN Wrapper module.

FlyingMonkey4 committed:
Merge branch 'dev' into TSATS-Util-Import

Koloss0 commented:
Hi @FlyingMonkey4! Sorry to burst your bubble, but uh... We just merged a new PR in TUK, so it would be worth updating to the new version before merging this PR.

Luckily, there aren't many differences. Here's the steps on how to do it:

1. Open your terminal and pull the latest version of TUK (See [Updating TUK](https://github.com/UMSATS/tsat-utilities-kit/wiki#updating))
2. The new version doesn't require a timer anymore, we got it to work with just RTOS. So you can fully disable it.
3. Re-compile and fix any syntax errors if there are any.
4. Once everything's in good condition, `cd` back to the CDH repository and push to your branch with the updated submodule.

It should be as easy as that. If you have any problems, let me or @ArnavG-it know!

FlyingMonkey4 committed:
updated submodules

updated tsat utilities kit and assured that the code compiles

FlyingMonkey4 committed:
added STM32L4 to preprocessing symbols

Koloss0 committed:
finalize TUK import

Changes:
 * Fixed some missing build settings in board and NUCLEO projects
 * Added RTOS to NUCLEO project
 * Deleted all references to the old can.h and can.h files.
 * Commented out some code that was no longer compatible with the new
   CAN Wrapper API. This will need to be fixed in the future.
 * Initialization of the CAN Wrapper Module in `main.c` of board
   project.

Both projects compile with TUK.
